### PR TITLE
[DOC] docstring for `SimpleRNNClassifier`

### LINKED ISSUE #4558: [DOC] write docstring for RNN estimators
The estimators `SimpleRNNRegressor` and `SimpleRNNClassifier` are practically empty, these should be added.

### PR DESCRIPTION
**Pull request by wasup-yash**
#### Reference Issues/PRs
Towards #4558

#### What does this implement/fix? Explain your changes.
Added DocStrings for the `[SimpleRNNClassifier]`(https://github.com/sktime/sktime/blob/main/sktime/classification/deep_learning/rnn.py#L20)

### CODE REVIEW

**Comment by yarnabrina** [comment]
@fkiraly just a quick question: why do most of `deep_learning` directory codes have a different docustring style from rest? These extra spaces before `:` are not really common in other `sktime` modules or even in `numpydoc` etc.

Not that it's objectively bad or anything, just asking because of inconsistent (if it matters for docstrings) style from most of other modules.

**Comment by fkiraly** [comment]
> @fkiraly just a quick question: why do most of `deep_learning` directory codes have a different docustring style from rest? These extra spaces before `:` are not really common in other `sktime` modules or even in `numpydoc` etc.

Because it has been without maintenance for a period of time - most deep learning estimators lived originally in `sktime-dl`, which got abandoned as its maintainers did not keep up with the professionalisation of the software architecture in the main package (which also included better docstrings as a side effect).

As content has been migrated to `sktime`, the interfaces have been updated, but the docstrings often still are in the old style.

File: sktime/classification/deep_learning/rnn.py [hunk 1]
```diff
@@ -19,6 +19,36 @@
 
 class SimpleRNNClassifier(BaseDeepClassifier):
     """Simple recurrent neural network.
+    
+    Parameters
+    ----------
+    should inherited fields be listed here?
```
Comment (fkiraly) [inline sktime/classification/deep_learning/rnn.py:25]: what do you mean by that? Like `random_state`?

Yes, all args of `__init__`.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks, this looks great!

Could you kindly remove the "should inherited fields" question from the docstring? Sth like this is better asked in the PR thread.

**Comment by wasup-yash** [comment]
> 

> Thanks, this looks great!
> 
> Could you kindly remove the "should inherited fields" question from the docstring? Sth like this is better asked in the PR thread.

ok I am gonna remove it, thanks for the review

**Review by fkiraly — CHANGES_REQUESTED** [review]
now waitasec, I just noticed - and apologies for not noticing earlier - the defaults in the docstring don't actually agree with those in the `__init__`!

Can you kindly fix that? Also, where the default is `None` (e.g., ,`optimizer`), you should check what it is set to internally, e.g., is it `Adam`?

**Comment by wasup-yash** [comment]
> now waitasec, I just noticed - and apologies for not noticing earlier - the defaults in the docstring don't actually agree with those in the `__init__`!
> 
> Can you kindly fix that? Also, where the default is `None` (e.g., ,`optimizer`), you should check what it is set to internally, e.g., is it `Adam`?

oh, sorry for not noticing it from my side too, I'll work on it and update it for the review.

**Review by fkiraly — APPROVED** [review]
Fixed outstanding items for the release

**Comment by wasup-yash** [comment]
> Fixed outstanding items for the release

thanks a lot @fkiraly for the fixes, I was indulged in uni exams and thought gonna work on it but you already fixed the docs, Really appreciate it.

**Pull request merged**